src/libasr/codegen/x86_assembler.h

```diff
@@ -465,6 +464,15 @@ class X86Assembler {
         return m_code;
     }
 
+    void align_by_byte(uint64_t alignment) {
+        uint64_t code_size = m_code.size() ;
+        uint64_t padding_size = (alignment * ceil(code_size / (double)alignment)) - code_size;
+        for (size_t i = 0; i < padding_size; i++) {
+            m_code.push_back(m_al, 0);
+        }
+        EMIT("times " + std::to_string(padding_size) + " db 0");
```

Comment:
**Line 473:** I think the `padding_size` computed here is valid only for the binary that we generate. For assembly text format, while encoding the `text_segment`, since `nasm` can encode instructions using different number of bytes than our encoding, the padding size required by `nasm` binary could be different from that of our binary. 

I am looking into the possibilities of computing the padding size dynamically in assembly text format.